features/build_runner/build_runner.rb

```diff
@@ -1,11 +1,185 @@
+      # TODO: better error handling, don't catch all Exception
+      puts(ex)
+      duration = Time.now - start_time
+      current_build.duration = duration
+      current_build.status = :failure # TODO: also handle failure
+      self.update_build_status!
+    end
+
     # @return [Array[String]] of references for the different artifacts created by the runner.
     def run(*args)
       not_implemented(__method__)
     end
+
+    # Responsible for updating the build status in our local config
+    # and on GitHub
+    def update_build_status!
+      update_build_status_locally!
+      update_build_status_source!
+    end
+
+    # Handle a new incoming row, and alert every stakeholder who is interested
+    def new_row(row)
+      logger.debug(row["message"])
+
+      # Report back the row
+      # 1) Store it in the history of logs (used to access half-built builds)
+      self.all_build_output_log_lines << row
+
+      # 2) Report back to all listeners, usually socket connections
+      self.build_change_observer_blocks.each do |current_block|
+        current_block.call(row)
+      end
+    end
+
+    # Add a listener to get real time updates on new rows (see `new_row`)
+    # This is used for the socket connection to the user's browser
+    def add_listener(block)
+      self.build_change_observer_blocks << block
+    end
+
+    def prepare_build_object
+      builds = Services.build_service.list_builds(project: self.project)
+
+      if builds.count > 0
+        new_build_number = builds.sort_by(&:number).last.number + 1
+      else
+        new_build_number = 1 # do we start with 1?
+      end
+
+      self.current_build = FastlaneCI::Build.new(
+        project: self.project,
+        number: new_build_number,
+        status: :pending,
+        timestamp: Time.now,
+        duration: -1,
+        sha: self.sha
+      )
+      update_build_status!
+    end
+
+    private
+
+    def update_build_status_locally!
+      # Create or update the local build file in the config directory
+      Services.build_service.add_build!(
+        project: self.project,
+        build: self.current_build
+      )
+
+      # Commit & Push the changes to git remote
+      FastlaneCI::Services.project_service.git_repo.commit_changes!
+    rescue StandardError => ex
+      logger.error("Error setting the build status as part of the config repo")
+      logger.error(ex.to_s)
+      logger.error(ex.backtrace.join("\n"))
+      # If setting the build status inside the git repo fails
+      # this is actually a big deal, and we can't proceed.
+      # For setting the build status, if that fails, it's fine
+      # as the source of truth is the git repo
+      raise ex
+    end
+
+    # Let GitHub know about the current state of the build
+    # Using a `rescue` block here is important
+    # As the build is still green, even though we couldn't set the GH status
+    def update_build_status_source!
+      return # TODO: enable again
```

Comment:
Remove this `return` once https://github.com/fastlane/ci/pull/230/files#r172956779 is resolved